pkg/apis/application/v1alpha1/types.go

```diff
@@ -1618,10 +1618,23 @@ func validateRoleName(name string) error {
 	return nil
 }
 
-var invalidChars = regexp.MustCompile("[,\n\r\t]")
+var invalidChars = regexp.MustCompile("[\"\n\r\t]")
 
 func validateGroupName(name string) error {
-	if strings.TrimSpace(name) == "" {
+	isQuoted := false
+	name = strings.TrimSpace(name)
+	if strings.HasPrefix(name, "\"") && strings.HasSuffix(name, "\"") {
+		isQuoted = true
+	}
+	if isQuoted {
+		// Remove surrounding quotes for further inspection of the group name
+		name = name[1 : len(name)-1]
+	} else {
+		if strings.Contains(name, ",") {
+			return status.Errorf(codes.InvalidArgument, "group '%s' must be quoted", name)
+		}
+	}
+	if name == "" {
```

Comment:
```suggestion
	name = strings.TrimSpace(name)
	if len(name) > 1 && strings.HasPrefix(name, "\"") && strings.HasSuffix(name, "\"") {
		// Remove surrounding quotes for further inspection of the group name
		name = name[1 : len(name)-1]
	} else if strings.Contains(name, ",") {
		return status.Errorf(codes.InvalidArgument, "group '%s' must be quoted", name)
	}

	if name == "" {
```

Added a string length check and simplified a bit.

Reply:
Thanks. How could I miss the string length check :facepalm:  Also, agree with the simplification.